docs/src/index.md

```diff
@@ -15,6 +15,9 @@ test data using functions that you don't need to precompile in your package.)
 Finally, `PrecompileTools` includes `@recompile_invalidations` to mitigate the undesirable consequences of *invalidations*.
 These different tools are demonstrated below.
 
+!!! note
+    There have been cases where precompilation led to performance regressions in packages (see for example [[1](https://github.com/JuliaLang/julia/issues/50749),[2](https://github.com/JuliaLang/julia/pull/50766),[3](https://github.com/JuliaLang/julia/issues/35972)]). While fixes have been released, some problems persist in older Julia versions and new issues might also slow down runtime after precompilation. Hence, it's advised to test packages for performance regressions in all supported Julia versions when introducing precompilation directives.
```

Comment:
```suggestion
    The latency reductions from PrecompileTools are maximally effective for Julia versions 1.9 and higher, and intermediate for Julia 1.8. Julia versions 1.7 and earlier may see some limited benefit as well, but have also occasionally been found to suffer from [precompilation-induced runtime performance regressions](https://github.com/JuliaLang/julia/issues/35972). If you wish, you can disable precompilation on older Julia versions by wrapping precompilation statements (see below) with `if Base.VERSION >= v"1.8" ... end`. On older Julia versions, you may wish to test packages for performance regressions when introducing precompilation directives.
```

I toned it down a bit; invoking the possibility that there could be unknown performance regressions, even though no one has reported any since https://github.com/JuliaLang/julia/pull/50766, seems a bit much. (We might as well say "using Julia theoretically exposes you to meltdown of your laptop; users are advised to install smoke-detectors near their desks.") I personally think we've nabbed them all, and if not Julia will just fix such regressions in Julia. https://github.com/JuliaLang/julia/issues/50749 doesn't apply to Julia versions before 1.9, so the only remaining case is exemplified by https://github.com/JuliaLang/julia/issues/35972.